docs/toolkits/openstack-automated-installation.md

```diff
@@ -0,0 +1,368 @@
+
+```shell
+root@ctl0:/home/deployer# ssh-keygen
+Generating public/private rsa key pair.
+Enter file in which to save the key (/root/.ssh/id_rsa): 
+Enter passphrase (empty for no passphrase): 
+Enter same passphrase again: 
+Your identification has been saved in /root/.ssh/id_rsa
+Your public key has been saved in /root/.ssh/id_rsa.pub
+The key fingerprint is:
+SHA256:95GBN6RU4eRJb19+NdpbKJXqTlz9xZtQH+uHG9Gf9ac root@ctl0
+The key's randomart image is:
++---[RSA 3072]----+
+|          ..*.   |
+|         . O o . |
+|          o B =+o|
+|           . B+BB|
+|        S . =o+o%|
+|         . + +oo&|
+|            = o**|
+|           o   +o|
+|            . E  |
++----[SHA256]-----+
+root@ctl0:/home/deployer#
+```
+
+Second, copy the resulting public key (stored in `/root/.ssh/id_rsa.pub` on ctl0) to the `/home/deployer/.ssh/authorized_keys` file on each of the other nodes. You can use the `ssh-copy-id` command to expedite this, e.g. with `ssh-copy-id deployer@compute0`.
+
+Third, make an initial SSH connection from the deployment node (root user) to each of the other node, e.g. with `ssh deployer@compute0`. When prompted, enter the password, and also accept the host key if prompted. This will establish a host key relationship so that Ansible can connect later.
+
+## Install Kolla Ansible
+
+### Install Dependencies
+
+On the deployment node, follow the steps in this section: <https://docs.openstack.org/kolla-ansible/yoga/user/quickstart.html#install-dependencies>
+
+From here on, use the "using a virtual environment" steps -- this avoids polluting the Python packages that are controlled by the system's package manager. Note that you'll need to replace `/path/to/venv` with a real filesystem path. This guide will use `/root/kolla-ansible-venv`.
+
+So, enter these commands:
+
+```bash
+apt update
+apt install python3-dev libffi-dev gcc libssl-dev
+apt install python3-venv
+python3 -m venv /root/kolla-ansible-venv
+source /root/kolla-ansible-venv/bin/activate
+pip install -U pip
+pip install 'ansible>=4,<6'
+```
+
+If you exit your shell, you must re-activate the virtual environment (run `source /root/kolla-ansible-venv/bin/activate`) before running many of the commands to follow.
+
+### Install Kolla Ansible itself
+
+Follow these steps on the deployment node: <https://docs.openstack.org/kolla-ansible/yoga/user/quickstart.html#install-kolla-ansible-for-deployment-or-evaluation>
+
+```
+pip install git+https://opendev.org/openstack/kolla-ansible@stable/yoga
+mkdir -p /etc/kolla
+chown root:root /etc/kolla
+cp -r /root/kolla-ansible-venv/share/kolla-ansible/etc_examples/kolla/* /etc/kolla
+cp /root/kolla-ansible-venv/share/kolla-ansible/ansible/inventory/* /etc/kolla
+```
+
+### Install Ansible Galaxy Requirements
+
+Follow these steps on the deployment node: <https://docs.openstack.org/kolla-ansible/yoga/user/quickstart.html#install-ansible-galaxy-requirements>
+
+```
```

Comment:
These are also similar to the previous.  Not sure if these commands need to be ran after or instead of the commands on the link.

Reply:
Fixed in latest commit